src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs

```diff
@@ -73,14 +73,14 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
             // Escape hatch to disable this resolver
             if (DisableNuGetSdkResolver.Value)
             {
-                return null;
+                return factory.IndicateFailure(null, new List<string>() { "NuGetSdkResolver did not resolve this SDK because it was disabled by the MSBUILDDISABLENUGETSDKRESOLVER environment variable." });
```

Comment:
Per coding guidelines, use named arguments on null arguments to clarify context.

```suggestion
                return factory.IndicateFailure(errors: null, warnings: new List<string>() { Strings.Error_DisabledSdkResolver });
```

Later, modify [Strings.resx](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/Strings.resx) file to include `Error_DisabledSdkResolver` string resource